src/subcommand/wallet/sign.rs

```diff
@@ -6,19 +6,33 @@ use {
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct Output {
   pub address: Address<NetworkUnchecked>,
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub inscription: Option<InscriptionId>,
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub output: Option<OutPoint>,
+  #[serde(skip_serializing_if = "Option::is_none")]
   pub message: Option<String>,
   pub witness: String,
 }
 
 #[derive(Debug, Parser)]
-#[clap(group(
+#[clap(
+group(
   ArgGroup::new("input")
     .required(true)
-    .args(&["message", "file"]))
+    .args(&["message", "file"])),
+group(
+  ArgGroup::new("signer")
+    .required(true)
+    .args(&["address", "inscription", "output"]))
 )]
 pub(crate) struct Sign {
   #[arg(long, help = "Sign for <ADDRESS>.")]
-  address: Address<NetworkUnchecked>,
+  address: Option<Address<NetworkUnchecked>>,
```

Comment:
Let's combine these into a `Signer` enum. We could either have this as an argument, or maybe a `--sign-with` flag.